pkg/workspace/library_loader.go

```diff
@@ -46,30 +46,36 @@ func NewLibraryLoader(libraryCtx LibraryExecutionContext,
 	}
 }
 
-func (ll *LibraryLoader) Schemas() ([]*yamlmeta.Document, error) {
+func (ll *LibraryLoader) Schemas() (Schema, error) {
 	loader := NewTemplateLoader(NewEmptyDataValues(), nil, ll.ui, ll.templateLoaderOpts, ll.libraryExecFactory, &schema.AnySchema{})
 
 	schemaFiles, err := ll.schemaFiles(loader)
 	if err != nil {
 		return nil, err
 	}
+	if ll.templateLoaderOpts.SchemaEnabled {
+		if len(schemaFiles) > 0 {
+			libraryCtx := LibraryExecutionContext{Current: schemaFiles[0].Library, Root: NewRootLibrary(nil)}
 
-	if len(schemaFiles) > 0 {
-		libraryCtx := LibraryExecutionContext{Current: schemaFiles[0].Library, Root: NewRootLibrary(nil)}
-
-		_, resultDocSet, err := loader.EvalYAML(libraryCtx, schemaFiles[0].File)
-		if err != nil {
-			return nil, err
-		}
+			_, resultDocSet, err := loader.EvalYAML(libraryCtx, schemaFiles[0].File)
+			if err != nil {
+				return nil, err
+			}
 
-		docs, _, err := DocExtractor{resultDocSet}.Extract(AnnotationSchemaMatch)
-		if err != nil {
-			return nil, err
+			docs, _, err := DocExtractor{resultDocSet}.Extract(AnnotationSchemaMatch)
```

Comment:
if we get multiple schema docs i would expect that we would blow up for now since we do not allow merging schemas.

Reply:
Correct, we have a story coming up to allow multiple schema docs #294. Prior to this story we would also blow up if we get multiple docs, so we haven't added guards for that case.